content/posts/golang/invalid-memory-address-or-nil-pointer-dereference.md

```diff
@@ -0,0 +1,170 @@
+---
+title: "Invalid Memory Address or Nil Pointer Dereference"
+summary: "Summary here"
+draft: true
+---
+
+## Overview {#overview}
+Example error:
+```golang
+panic: runtime error: invalid memory address or nil pointer dereference
+[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x497783]
+```
+
+This issue happens when you reference an unassigned (nil) pointer.
+
+## Initial Steps Overview {#initial-steps-overview}
+
+1) [Assign a variable to the pointer](#step-1)
+
+2) [Use a direct reference instead of a pointer](#step-2)
+
+3) [Check for nil assignments being made to pointers](#step-3)
+
+## Detailed Steps {#detailed-steps}
+
+### 1) Assign a variable to the pointer {#step-1}
```

Comment:
These are solutions. How do I identify where the problem is, and which solution to apply?